src/irmin-pack/irmin_pack.ml

```diff
@@ -448,6 +448,9 @@ struct
         Contents.CA.close (contents_t t) >>= fun () ->
         Node.CA.close (snd (node_t t)) >>= fun () ->
         Commit.CA.close (snd (commit_t t)) >>= fun () -> Branch.close t.branch
+
+      (** stores share on-disk files, one ro_sync is enough *)
```

Comment:
It's not clear to me why this is true. Don't these store instances contain separate mutable state that needs to be updated? (albeit to the same values in each case.) For instance, will the `Node` instance detect a change to the generation number?

Reply:
the three stores share the same instance of both pack (https://github.com/mirage/irmin/blob/master/src/irmin-pack/pack.ml#L145) and dict (https://github.com/mirage/irmin/blob/master/src/irmin-pack/pack_dict.ml#L18), if I understand this cache correctly.

Reply:
I certainly don't understand it correctly, so I defer to your judgement :slightly_smiling_face:

Reply:
Thinking about this a bit more, I think this is good but the comment might need changed:

```suggestion
      (** stores share instances in memory, one ro_sync is enough *)
```

i.e. it's not safe because just the files are the same, it's safe because – in addition – the in-memory mutable state that tracks those fails is identical.